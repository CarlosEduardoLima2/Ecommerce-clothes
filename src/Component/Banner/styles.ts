import styled, { css } from 'styled-components';
import {Container as TextComponent} from '../TextComponent/styles'
import {Container as Heading} from '../Heading/styles'


export const Container = styled.section`
  ${({theme}) => css`
    width: 100%;
    height: 100vh;
    background-color: ${theme.colors.primaryColor};

    @media (max-width: 880px){
    height: 100%;
    }
  `}
`;

export const Content = styled.div`
  ${({theme}) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 16px;
    gap: 2rem;

    .swiper-slide {
      display: flex;
      justify-content: center;
      background: #fff;
      background: transparent;
      align-items: center;


      div{
        margin: 30px 0;
      }

      ${TextComponent}{
        display: block;
        max-width: 500px;
        margin: 15px 0;
        font-size: 0.8rem;
      }

    }

    @media (max-width: 460px) {
      flex-wrap: wrap;
    }
    
  `}
`;

export const Text = styled.section`
  
`;
export const Image = styled.div`
  ${({theme}) => css`
    max-width: 350px;
    max-height: 600px;
    img{
      object-fit: cover;
      object-position: center;
      height: 100%;
    }

    @media (max-width: 880px){
      img{
        width: 80%;
        height: 80%;
      }
    }
  `}
`;