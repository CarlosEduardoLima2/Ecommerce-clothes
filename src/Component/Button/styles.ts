
import styled, { css } from 'styled-components';
import { NavLink } from 'react-router-dom';

export const Container = styled(NavLink)`
  ${({theme}) => css`
    padding: 10px 60px;
    border-radius: 8px;
    color: ${theme.colors.lightColor};
    background-color: ${theme.colors.darkColor};
    text-decoration: none;
    transition: .3s ease-in;

    &:hover{
      border: none;
      background: #363636;
    }
  `}
    
`;
