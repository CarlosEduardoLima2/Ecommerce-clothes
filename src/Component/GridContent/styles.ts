import styled from 'styled-components';

export const Container = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit,minMax(250px,1fr));
  gap: 88px 30px;
`;
