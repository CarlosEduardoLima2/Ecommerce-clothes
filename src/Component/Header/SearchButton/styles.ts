import styled from 'styled-components';

export const Container = styled.button`
  border: none;
  background-color: #fff;
  transition: .4s ease;

  &:hover{
    opacity: 0.6;
  }
`;
