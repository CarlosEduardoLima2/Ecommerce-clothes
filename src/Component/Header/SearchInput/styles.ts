import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    background-color: #fff;
    border-bottom: 3px solid #000;
  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    width: 100%;
    background-color: ${theme.colors.lightColor};
    border: none;
    font-size: ${theme.fonts.sizes.xmedium};
    padding: 15px 30px;
    color: ${theme.colors.darkColor};
    font-weight: 700;

    &:focus {
      outline: 0;
    }
  `}
`;
