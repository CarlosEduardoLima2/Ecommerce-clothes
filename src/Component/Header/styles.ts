import styled, { css } from 'styled-components';

export const Container = styled.header`
  z-index: 6;
  width: 100%;
  background: transparent;
`;

export const Header = styled.div`
  ${({theme}) => css`
    padding: 30px 0;
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
  `}
  
`;

export const Nav = styled.div`
  display: flex;
  grid-column-gap: 1.2rem;
  align-items: center;
`;

export const Logo = styled.div`

`;

export const Actions = styled.div`

`;
