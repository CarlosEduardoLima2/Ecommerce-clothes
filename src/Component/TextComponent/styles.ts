import styled, { css, DefaultTheme } from 'styled-components';
import { TextComponentProps } from '.';

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const background = (theme: DefaultTheme) => css`
  background-color: ${theme.colors.secondaryColor};
  width: fit-content;
  width: -webkit-fit-content;
  font-size: ${theme.fonts.sizes.small};
  font-weight: 700;
  padding: 12px 20px 10px;
  font-style: normal;
  border-radius: ${theme.fonts.sizes.xsmall};
`

export const Container = styled.div<TextComponentProps>`
  ${({ theme, uppercase, hasBackground }) => css`
    color: ${theme.colors.darkColor};
    font-style: italic;
    ${uppercase && titleCase(uppercase)};
    ${hasBackground && background(theme)}
  `}
`;
