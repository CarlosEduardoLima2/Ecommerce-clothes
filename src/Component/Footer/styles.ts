
import styled from 'styled-components';
import {Container as Text} from "../TextComponent/styles"
import {Container as Link} from "../NavLinkComponent/styles"
import { theme } from '../../styles/theme';

export const Container = styled.div`
  background-color: ${theme.colors.primaryColor};
  width: 100%;
`;

export const Info = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-wrap: wrap;

  @media ${theme.media.lteMedium} {
    justify-content: center;
  }
`;

export const DataContainer = styled.div`
  padding: 1.5rem 0;

  ${Text}{
    margin: 1rem 0;
  }
`;

export const Icons = styled.div`

`;
export const InfoText = styled.div`

`;

export const LinksContainer = styled.div`
  display: flex;
  flex-direction: column;

  ${Link}{
    padding: 0.5rem;
  }
`;

export const About = styled.div`
  max-width: 300px;
  padding: 1rem;
  text-align: center;
  ${Text}{
    margin: 1rem 0;
  }
`;
