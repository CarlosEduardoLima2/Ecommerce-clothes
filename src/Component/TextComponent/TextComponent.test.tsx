import { screen } from '@testing-library/react';
import { TextComponent } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<TextComponent />', () => {
  it('should render with default values', () => {
    renderTheme(<TextComponent>children</TextComponent>);
    const text = screen.getByText('children');
    expect(text).toHaveStyle({
      color: theme.colors.darkColor,
      'font-style': 'italic',
    })
  });

  it('should render text background', () => {
    renderTheme(<TextComponent hasBackground>children</TextComponent>);
    const text = screen.getByText('children');
    expect(text).toHaveStyle({
      'background-color': theme.colors.secondaryColor,
      width: 'fit-content',
      'font-size': theme.fonts.sizes.small,
      'font-weight': '700',
      padding: '12px 20px 10px',
      'font-style': 'normal',
      'border-radius': theme.fonts.sizes.xsmall,
    })
  });

  it('should render with uppercase letters', () => {
    renderTheme(<TextComponent uppercase>children</TextComponent>);
    const text = screen.getByText('children');
    expect(text).toHaveStyle({
      'text-transform': 'uppercase',
    })
  });

  it('should mach snapshot', () => {
    const { container } = renderTheme(<TextComponent>texto</TextComponent>);

    expect(container).toMatchSnapshot();
  });
});
