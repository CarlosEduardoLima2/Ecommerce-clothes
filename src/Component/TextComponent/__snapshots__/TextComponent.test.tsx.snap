// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> should mach snapshot 1`] = `
.c0 {
  color: #212121;
  font-style: italic;
}

<div>
  <div
    class="c0"
  >
    texto
  </div>
</div>
`;
